src/main/java/net/geforcemods/securitycraft/entity/SentryEntity.java

```diff
@@ -219,16 +246,29 @@ else if(item == SCContent.UNIVERSAL_OWNER_CHANGER.get())
 	}
 
 	/**
-	 * Cleanly removes this sentry from the world, dropping the module and removing any block
+	 * Cleanly removes this sentry from the world, dropping the module and removing the block the sentry is disguised with
 	 */
 	@Override
 	public void remove()
 	{
+		BlockPos pos = getPosition();
+		if (!getDisguiseModule().isEmpty())
+		{
+			List<Block> blocks = ((ModuleItem)getDisguiseModule().getItem()).getBlockAddons(getDisguiseModule().getTag());
+
+			if(blocks.size() > 0)
+			{
+				Block block = blocks.get(0);

```

Comment:
Don't really need this variable here. You can just call `blocks.get(0)` directly in the if condition below.